common/lifecyleMgr.go

```diff
@@ -515,13 +524,6 @@ func (lcm *lifecycleMgr) InitiateProgressReporting(jc WorkController) {
 				}
 			}
 
-			// wait a bit before fetching job status again, as fetching has costs associated with it on the backend
-			select {
-			case <-lcm.cancelChannel:
-				doCancel()
-			case <-time.After(wait):
```

Comment:
If I'm understanding correctly, we are now doing a busy wait?

Have you considered giving a minimum wait period (like 10ms) so that we don't busy wait?

Reply:
Yes, we're doing a busy wait. I'll add a select channel instead of busy wait here.